state/upgrades.go

```diff
@@ -1930,6 +1930,11 @@ func updateKubernetesStorageConfig(st *State) error {
 	if err != nil || model.Type() == ModelTypeIAAS {
 		return errors.Trace(err)
 	}
+	if model.Life() != Alive {
```

Comment:
I think that this is where we want to be overall in the long run... However, there are some IAAS models that are already dying on older Juju versions and can only be removed if they are upgraded to newer Juju versions... I wonder if we should be only putting this here for CAAS models.

Reply:
Oh, this whole step is only for caas models - it exits early for iaas ones.